src/serialize.h

```diff
@@ -371,38 +372,73 @@ template<typename Stream, VarIntMode Mode, typename I>
 void WriteVarInt(Stream& os, I n)
 {
     CheckVarIntMode<Mode, I>();
-    unsigned char tmp[(sizeof(n)*8+6)/7];
-    int len=0;
-    while(true) {
-        tmp[len] = (n & 0x7F) | (len ? 0x80 : 0x00);
-        if (n <= 0x7F)
+    switch(Mode) {
+        case VarIntMode::BITSTREAM: {
+            unsigned char size = (sizeof(n)*8)-std::__countl_zero(n);
+            if (!size) { ser_writedata8(os, 0); break; }
+            unsigned char chunks = ceil(size/7.0);
```

Comment:
In ec2b2ea56c06bf72e53b0ecfb4065ccf2bd09cb0:

Don't use `ceil(size/7.0)` like this. It's slow and unidiomatic. Use `(size + 6) / 7`. And why are you using `unsigned char` here for numeric values? It seems likely to overflow.

Reply:
Fixed, Thanks!